presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestSwapAdjacentWindowsBySpecifications.java

```diff
@@ -125,14 +125,19 @@ public void dependentWindowsAreNotReordered()
                         p.window(new WindowNode.Specification(
                                         ImmutableList.of(p.symbol("a")),
                                         Optional.empty()),
-                                ImmutableMap.of(p.symbol("avg_1"),
-                                        new WindowNode.Function(new FunctionCall(QualifiedName.of("avg"), windowA, false, ImmutableList.of(new SymbolReference("avg_2"))), functionHandle, frame)),
+                                ImmutableMap.of(p.symbol("avg_1"), newWindowNodeFunction(ImmutableList.of(new SymbolReference("avg_2")))),
                                 p.window(new WindowNode.Specification(
                                                 ImmutableList.of(p.symbol("a"), p.symbol("b")),
                                                 Optional.empty()),
-                                        ImmutableMap.of(p.symbol("avg_2"),
-                                                new WindowNode.Function(new FunctionCall(QualifiedName.of("avg"), windowA, false, ImmutableList.of(new SymbolReference("a"))), functionHandle, frame)),
+                                        ImmutableMap.of(p.symbol("avg_2"), newWindowNodeFunction(ImmutableList.of(new SymbolReference("a")))),
                                         p.values(p.symbol("a"), p.symbol("b")))))
                 .doesNotFire();
     }
+
+    private WindowNode.Function newWindowNodeFunction(List<SymbolReference> symbols)
```

Comment:
Why not take a list of `VariableReferenceExpression` directly? Or a list of `Symbol` even.